pkg/runtime/host.go

```diff
@@ -8,25 +8,37 @@ import (
 )
 
 const (
+	// HostIPEnvVar is the environment variable to override host's chosen IP address.
 	HostIPEnvVar = "DAPR_HOST_IP"
 )
 
+// GetHostAddress selects a valid outbound IP address for the host.
 func GetHostAddress() (string, error) {
 	if val, ok := os.LookupEnv(HostIPEnvVar); ok && val != "" {
 		return val, nil
 	}
 
-	addrs, err := net.InterfaceAddrs()
+	// Use udp so no handshake is made.
+	// Any IP can be used, since connection is not established, but we used a known DNS IP.
+	conn, err := net.Dial("udp", "8.8.8.8:80")
```

Comment:
How much delay if its unreachable or slow network or if app is deployed in a constrained/locked down environment (gov. cloud regions, some teams dont allow any external traffic outside of their vNET)? I am concerned about delay in startup in production environment.

Reply:
Might want to introduce a `acceptable-delay-ms` here.

acceptable-delay is the time in ms that after checking, we think is the reasonable amount of time to wait for a response.

Reply:
I would check what's the average response time for this command is on k8s, and then construct the timeout from that.

Reply:
There is no connection. No packet leaves the machine here. This is just logical.

Reply:
See https://stackoverflow.com/questions/23558425/how-do-i-get-the-local-ip-address-in-go

There is a solution from Marcel Molina with 96 votes (as of now) and there is a discussion about this there.

Reply:
@artursouza  I believe, the Dial function connects to server and it wont be able to do that in a constrained environment eventually timing out and causing delay

Reply:
@amanbha This is UDP, there is no connection. I am happy to do an experiment to clarify this. No packet leaves the host on "Dial".

Reply:
Replay experiment here: https://play.golang.org/p/dVD9ApVBqH5

Reply:
This other one is similar but displays the IP selected like in this code: https://play.golang.org/p/gDt0IcbkC1A

Reply:
Let me know is there is any concerns with this PR.